dev-utilities/release-utilities/nifi-rc-check

```diff
@@ -0,0 +1,326 @@
+    #
+    #   -c=RC2
+    #   --candidate=RC2
+    #
+    -c=*|--candidate=*)
+    CANDIDATE="${i#*=}"
+    shift # past argument=value
+    ;;
+
+    #
+    # -h/--help
+    #
+    -h|--help)
+    help
+    exit 0
+    shift # past argument with no value
+    ;;
+
+    #
+    # Unknown option
+    #
+    *)
+    UNKNOWN_OPTION="${i#*=}"
+    echo "Error: unknown option: $UNKNOWN_OPTION"
+    help
+    ;;
+  esac
+done
+
+# validation
+if [ -z "$VERSION" ]; then
+	echo "Missing -v/--version is is required"
+	exit 1
+fi
+if [[ "$VERSION" =~ ^[0-9]{1,2}\.[0-9]{1,2}\.[0-9]{1,2} ]]; then
+  NIFI_VERSION="$VERSION"
+else
+  print_error "$VERSION may not be a valid version number"
+  exit 1
+fi
+
+if [ -z "$CANDIDATE" ]; then
+	echo "Missing -c/--candidate which is required"
+	exit 1
+fi
+
+if [[ "$CANDIDATE" =~ ^RC[0-9]+ ]]; then
+  RC=$(echo "$CANDIDATE" | tr '[:upper:]' '[:lower:]')
+  UPPER_RC=$(echo "$CANDIDATE" | tr '[:lower:]' '[:upper:]')
+elif [[ "$CANDIDATE" =~ ^[0-9]+ ]]; then
+  RC=rc"$CANDIDATE"
+  UPPER_RC=RC"$CANDIDATE"
+else
+  print_error "invalid RC, valid is RC# or just #"
+  exit 1
+fi
+
+print_section_header "Apache Nifi Version $NIFI_VERSION $UPPER_RC"
+
+NIFI_REL_RC="$NIFI_NAME_LOWER-$NIFI_VERSION-$UPPER_RC"
+
+NIFI_RC_DIST="$NIFI_DEV_DIST$NIFI_NAME_LOWER-$NIFI_VERSION/"
+print_item "Apache Nifi RC Distribution Root is $NIFI_RC_DIST"
+
+
+# working directory
+WORK="$HOME/tmp/nifi-release-$NIFI_VERSION"
+
+# handle tilde expansion
+WORK="${WORK/#\~/$HOME}"
+
+# warn the user if the working directory exists
+if [ -d "$WORK" ]; then
+  print_error "Directory $WORK exists, please rename it and start over"
+  exit 1
+fi
+
+if [ ! -d "$WORK" ]; then
+  mkdir -p "$WORK"
+fi
+print_item "Working directory is $WORK"
+
+print_section_header "Validation"
+
+KEYS="$NIFI_DEV_DIST/KEYS"
+LOCAL_ASSEMBLY="$NIFI_NAME_LOWER-$NIFI_VERSION-source-release.zip"
+NIFI_ASSEMBLY="$NIFI_RC_DIST$NIFI_NAME_LOWER-$NIFI_VERSION-source-release.zip"
+NIFI_ASSEMBLY_SIG="$NIFI_ASSEMBLY.asc"
+NIFI_ASSEMBLY_SHA1="$NIFI_ASSEMBLY.sha1"
+NIFI_ASSEMBLY_SHA256="$NIFI_ASSEMBLY.sha256"
+NIFI_ASSEMBLY_SHA512="$NIFI_ASSEMBLY.sha512"
+
+print_section_item "Downloading $KEYS"
+if ! wget -P "$WORK" "$KEYS" ; then
+  print_error "Failed to download $KEYS"
+  exit 1
+fi
+
+print_section_item "Downloading $NIFI_ASSEMBLY"
+if ! wget -P "$WORK" "$NIFI_ASSEMBLY" ; then
+  print_error "Failed to download $NIFI_ASSEMBLY"
+  exit 1
+fi
+print_section_item "Downloading $NIFI_ASSEMBLY_SIG"
+if ! wget -P "$WORK" "$NIFI_ASSEMBLY_SIG" ; then
+  print_error "Failed to download $NIFI_ASSEMBLY_SIG"
+  exit 1
+fi
+
+cd "$WORK" || exit 1
+print_section_item "importing nifi keys"
+
+if ! gpg --import KEYS ; then
+  print_error "failed to import KEYS"
+  exit 1
+fi
+
+print_section_item "Verifying Nifi Assembly"
+if ! gpg --verify ./"nifi-$NIFI_VERSION-source-release.zip.asc" "nifi-$NIFI_VERSION-source-release.zip" ; then
+  print_error "failed to verify Nifi Assembly"
```

Comment:
This should use `gpg --verify -v ...` to ensure that the internal digest algorithm being used for the signature is SHA-512.